torch/utils/data/dataloader.py

```diff
@@ -65,6 +67,15 @@ class DataLoader(object):
 
     .. warning:: If ``spawn`` start method is used, :attr:`worker_init_fn` cannot be an
                  unpicklable object, e.g., a lambda function.
+
+    .. warning:: The default memory pinning logic only recognizes Tensors and maps and iterables
+                 containg Tensors.  By default, if the pinning logic sees a batch that is a custom type,
+                 (which will occur if you have a ``collate_fn`` that returns a custom batch type),
+                 or if each element of your batch is a custom type, the pinning logic will not
+                 recognize them, and it will return that batch (or those elements)
+                 without pinning the memory.  To enable memory pinning for custom batch or data types,
+                 define a pin_memory method on your custom type(s).  See ``SimpleCustomBatch`` and
```

Comment:
Users don't have direct access to these files. I would prefer writing them in an `Example::` code block below.

Reply:
Ok, a minimal-but-complete in-place example doesn't actually look too bad there imo.  Let me know what you think.